app/components/Wallet/SuiWallet/useBalance.tsx

```diff
@@ -17,43 +17,59 @@ export function useCoinBalance(coinAddr?: string): UseCoinBalanceResult {
 	const [balance, setBalance] = useState<CoinBalance | null>(null);
 	const [isLoading, setIsLoading] = useState<boolean>(true);
 	const [error, setError] = useState<Error | null>(null);
+	// Keep the fetching logic simple and reusable
+	const fetchBalance = useCallback(
+		async (owner: string, coin: string | undefined) => {
+			setIsLoading(true);
+			setError(null);
+
+			let cancelled = false;
+			try {
+				const result = await suiClient.getBalance({ owner, coinType: coin });
+				if (!cancelled) {
+					setBalance(result);
+				}
+			} catch (err) {
+				if (!cancelled) {
+					setError(err instanceof Error ? err : new Error("Failed to fetch balance: " + err));
+					setBalance(null);
+				}
+			} finally {
+				if (!cancelled) {
+					setIsLoading(false);
+				}
+			}
+
+			return () => {
+				cancelled = true;
+			};
+		},
+		[suiClient],
+	);
```

Comment:
We already have similar funciton. I think we are doing a dupliaciton here. Let's see if we can reuse it. Check
```
~/components/Wallet/SuiWallet/useBalance
```